src/slowmokit/methods/activation_functions.cpp

```diff
@@ -0,0 +1,130 @@
+/**
+ * @file methods/activation_functions.cpp
+ *
+ * Implementation of activation functions
+ */
+#include "activation_functions.hpp"
+template<class T>
+// sigmoid
+std::vector<double> sigmoid(const std::vector<double> &x)
```

Comment:
> Instead of initialising a new vector and returning it....change the return type to `void` and do the changes in vector `x` itself wherever possible

done